components/dashboard/src/settings/Preferences.tsx

```diff
@@ -19,70 +18,57 @@ import settingsMenu from "./settings-menu";
 
 type Theme = 'light' | 'dark' | 'system';
 
+const DesktopNoneId = "none"
+const DesktopNoneLogoDark = "image/none-dark.svg"
+const DesktopNoneLogoLight = "image/none.svg"
+const DesktopNone: IDEOption = {
+    "image": "",
+    "logo": DesktopNoneLogoLight,
+    "orderKey": "-1",
+    "title": "None",
+    "type": "desktop"
+}
+
 export default function Preferences() {
     const { user } = useContext(UserContext);
-    const { setIsDark } = useContext(ThemeContext);
-
-    const [defaultIde, setDefaultIde] = useState<string>(user?.additionalData?.ideSettings?.defaultIde || "");
-    const actuallySetDefaultIde = async (value: string) => {
-        const additionalData = user?.additionalData || {};
-        const settings = additionalData.ideSettings || {};
-        settings.defaultIde = value;
+    const { setIsDark, isDark } = useContext(ThemeContext);
+
+    const updateUserIDEInfo = async (defaultDesktopIde: string, defaultIde: string) => {
+        let useDesktopIde = defaultDesktopIde !== DesktopNoneId;
+        const additionalData = user?.additionalData ?? {};
+        const settings = additionalData.ideSettings ?? {};
```

Comment:
Whoa, I had to read [MDN: Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) again in order to understand this. 😅

**Suggestion**: I think using the nullish coalescing operator is not necessary here, as a simple `||` can do the job (and is much more common and readable):

```suggestion
        const additionalData = user?.additionalData || {};
        const settings = additionalData.ideSettings || {};
```

Reply:
Nullish coalescing operator `??` is for disambiguation, we should give preference to `??`
Here we use `??`, means that if `user?.additionalData` is null we got a default value `{}`

eg:
```
// data = { boolValue: false }
let value = data?.boolValue || true     // true
let value = data?.boolValue ?? true.  // false
```

Reply:
Indeed, this is correct for number and booleans, but here `additionalData` and `ideSettings` should always be objects.

The only case where `??` and `||` have a different behavior is if `additionalData` has a non-object, non-null value, e.g. `false` or `''`. These cases are theoretically impossible, but even if they happen, I think we'd still prefer setting the default value of `{}` instead of keeping the bad value `false` or `''`.

My point is that `??` is a rare feature, which might hurt the readability and maintainability of the dashboard code.

However, I don't have a very strong opinion, so if you insist on keeping the `??` here, I won't block your valuable Pull Request for that. 😇

Reply:
Yes, I insist my option. It's fine here to use `||`, but with `??` we don't need to think about, what `additionalData` is, is it a bad zero value for this expression or not?